framework/packages/abstract-client/src/account.rs

```diff
@@ -160,18 +160,73 @@ impl<Chain: CwEnv> Account<Chain> {
     // Install an application on the account
     // creates a new sub-account and installs the application on it.
     pub fn install_app<
-        M: ContractInstance<Chain> + RegisteredModule + From<Contract<Chain>> + Clone,
+        M: ContractInstance<Chain> + InstallConfig + From<Contract<Chain>> + Clone,
     >(
         &self,
         configuration: &M::InitMsg,
         funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        self.install_app_internal(vec![M::install_config(configuration)?], funds)
+    }
+
+    pub fn install_app_with_dependencies<
+        M: ContractInstance<Chain>
+            + DependencyCreation
+            + InstallConfig
+            + From<Contract<Chain>>
+            + Clone,
+    >(
+        &self,
+        module_configuration: &M::InitMsg,
+        dependencies_config: M::DependenciesConfig,
+        funds: &[Coin],
+    ) -> AbstractClientResult<Application<Chain, M>> {
+        let mut install_configs: Vec<ModuleInstallConfig> =
+            M::dependency_install_configs(dependencies_config)?;
+        install_configs.push(M::install_config(module_configuration)?);
+
+        self.install_app_internal(install_configs, funds)
+    }
+
+    pub fn admin(&self) -> AbstractClientResult<Addr> {
+        self.abstr_account.manager.address().map_err(Into::into)
```

Comment:
The Admin of the account is not the manager, the **direct** admin of a module is the manager, the admin of the account / indirect admin of a module is the owner of the account. 

`.owner` or `.admin` should return the address that controls the account. For sub-accounts that might get tricky (do we return the proxy of the owning account or the top-level owner of all the accounts?) (related, do we want a `SubAccount` struct or not?)

I think an application should have a `.direct_admin` that returns the manager and `.account_admin` or `.account_owner` that returns the obvious

Reply:
Do we not want to pass along the `manager`/`proxy` naming through the client? I think as a consumer `direct_admin` and `account_admin` might be more confusing.

Reply:
I already added a `proxy` query earlier it seems. I think it would be nice to keep `manager` and `proxy` for consistency with our docs.

Reply:
Well, with the sub-accounts `direct_admin` is not an account's admin. The direct admin of any account is either a wallet/multisig/whatever or another account.

Reply:
So it returning the manager address doesn't make any sense imo

Reply:
Is how it is now incorrect (see new commits)? I thought you said `direct_admin` was manager and `account_owner` was the owner: 

>I think an application should have a .direct_admin that returns the manager and .account_admin or .account_owner that returns the obvious

Reply:
For an `Application`, this code is for an `Account`